pkg/agent/types/knapsack.go

```diff
@@ -0,0 +1,30 @@
+package types
+
+import "go.etcd.io/bbolt"
+
+// Knapsack is an inventory of data and useful services which are used throughout
+// launcher code and are typically valid for the lifetime of the launcher application instance.
+type Knapsack struct {
+	Storage Storage
+
+	// BboltDB is the underlying bbolt database.
+	// Ideally, we can eventually remove this. This is only here because some parts of the codebase
+	// like the osquery extension have a direct dependency on bbolt and need this reference.
+	// If we are able to abstract bbolt out completely in these areas, we should be able to
+	// remove this field and prevent "leaking" bbolt into places it doesn't need to.
+	BboltDB *bbolt.DB
+
+	// This struct is a work in progress, and will be iteratively added to as needs arise.
+	// Some potential future additions include:
+	// Flags
+	// Querier
+}
+
+func NewKnapsack(s Storage, db *bbolt.DB) *Knapsack {
```

Comment:
Suggestion ... 

We might consider putting Knapsack in it's own pkg. Possibly at `pkg/agent/knapsack`. If we think this is a good idea, It could wait until next PR.

If this will become truly ubiquitous, I can imagine there could be several convenience constructors / options we would use in tests or other situations.

`knapsack.New(WithMemStore, WithFmtLogger, WithFlagFoo, ...)`

Reply:
I don't mind doing this now, I have a feeling it will become pervasive enough that doing this now is worth it. Good suggestion!